### DISCUSSION TITLE
HCIBox feature brainstorming

### DISCUSSION THREAD

**Discussion by dkirby-ms** [microsoft/azure_arc#1454] [Ideas]
What are some things that the hybrid community would like to see included in future HCIBox releases?

Some ideas we've been considering:

- SDN support with hybrid AKS
- Expanded sample applications
- Azure IoT integration
- Azure Arc-enabled data services with SQL MI 
- More flexible deployment configurations

Share and discuss your ideas for HCIBox features here!

**Comment by realjonas** [comment]
To be able to deploy HCIBox onprem, on Hyper-v or bare metal, would be awesome!!!!!
/jonas

**Reply by mgodfre3** [reply]
Would this be on a single Windows Server running Hyper-V, so you could have a GUI method of control, but also have the sandbox isolated from your current physical network? Or are you looking for a solution that runs on one or more bare metal hosts, running Azure Stack HCI?

**Reply by realjonas** [reply]
Hi,
My first thought was to run on a windows server running hyper-v to have a sandbox environment for training and to save some money with not be charging for the azure nested vm. Your second question would be cool but I guess that it is just what azure stack edge pro 2 is, in some way.

For the windows server hyper-v scenario I am aware of that I can use, and I have done, the MSlab hydration kit, but it would be much cooler to able to hydrate like the bicep version of Arc jumpstart HCI box.
Br,
//Jonas

**Comment by dkirby-ms** [comment]
Technically, you could take your Windows server hyper-v host and run the bootstrap script on it to prep, and then run HCIBoxLogonScript and in theory that would give you the same as the regular HCIBox automation.

It hasn't been tested this way though, and we have been discussing options for providing a "bring your own computer" version of HCIBox. Nothing more concrete to share at this time but its worth further discussion.

**Reply by realjonas** [reply]
Hi!
Thank's
Yes I have thought about that and started to doing it that way.
anyway awesome job you guy's have done so far!
Have a nice weekend!
/Jonas

**Reply by likamrat** [reply]
@realjonas would be great to hear about your findings and experience with your hardware if that is something you plan on testing

**Reply by realjonas** [reply]
Hi,

Finally, I managed to bring my own computer, into a ArcHCIBox Jumpstart.
I chose to run it feature by feature so I could fix bugs and make modifications along the way.

I started by changing and commenting out the services I wanted to create on-prem in the bicep files, then I deployed the Azure resources with Bicep.

Then it was time to go to my lab server, a collection of an old Fujitsu Primary RX200 S8 server with 1 Gbit network cards, 256GB ram, 2 TB SSD storage and Windows 2022 Datacenter.

I Ran the bootstrap script on the server and then stepped through the logon script step by step, I ran it section by section.

Hade some trouble the first time, but after some modifications it worked well.

Happy to have a local Jumpstart HCIBox! :-)

Next step is to try deploying the SDN stuff, network controller etc. And after that see if I can get it to run in an more automated way, I have a thought to try use MDT to netboot baremetal and with a task sequence auto deploy Jumpstart HCIBox.

**Changes I did:**
1.	Modified the Subnets to match my local settings
2.	Changed Mtu size to 1514 in HCIBox-Config.psd1 file
3.	In the function Add-Files, in file “New-HCIBoxCluster.ps1”, Changed X to Z mount point at the # Mount VHDX section for 
        AzSMGMT Vm.

**Steps process:**
1.	Deployed Azure resources with modified Bicep files
2.	Bootstrap.ps1 on local server
3.	HCIBoxLogonScript.ps1 local server
             a) New-HCIBoxCluster.ps1
             b) Register-AzSHCI.ps1
             c) Deploy-AKS.ps1
             d) Deploy-ArcResourceBridge.ps1

**Comment by dkirby-ms** [comment]
Updating this conversation with some news - work is nearly finished on supporting 23H2. This will also resolve the current outstanding resource bridge issues which are related to the changes to how RB is deployed.

We will be releasing support for 23h2 before adding any other capabilities like richer SDN support, etc. Other top requests include:

* Offline mode (run on your own hardware)
* single node option
* SDN examples and support
* AVD on HCI example

Share other ideas you have here.

**Reply by dkirby-ms** [reply]
Another thing to mention is that deploying the actual cluster resource to Azure will be different with HCIBox 23h2 support. For the short term, it will mean a manual cluster deployment via ARM template by the user. Once CLI cluster deployment is supported the cluster can be deployed via scripted automation as it does today for 22h2.

**Reply by fberson** [reply]
Hi @dkirby-ms, any news on the 23H2 support? Looking forward to deploying this!

**Reply by jvhoof** [reply]
Is SDN controller support in the current version of HCIbox? Are there already some additional SDN examples? Tried to deploy but received very quickly a deployment error regarding an incorrect parameter.
Great resource to test Azure Stack HCI!

**Comment by dkirby-ms** [comment]
Hi @fberson, I just updated the issue with the latest but posting here for additional visibility.

I have seen repeated validation errors when deploying with the current 2310 build of AzSHCI. I was advised to wait until the 2311 release (which should be coming in the next few days) to have the validation issues resolved. Otherwise, the hcibox_23h2 branch I have in my fork is ready to go with the updates for 23h2 along with a number of other code revisions.

**Comment by hemetz** [comment]
please provide another much faster source to download the HCIBox images. The script deploys the environment in around ~25min, but the post-deployment process requires more than 4 hours based on the extrem slow download speed for the HCIbox images:


Maybe it could help when you add "-Priority 'Foreground'" to the BITSdownload cmd at azure_jumpstart_hcibox/artifacts/PowerShell/New-HCIBoxCluster.ps1 line 19.

but I think the source is the issue. When I try to directly download the source images via browser from aka.ms/VHD-HCIBox-HCI-Prod and https://aka.ms/VHD-HCIBox-Mgmt-Prod i receive the same spikes as via BITS and no continuous download ![332710146-0157322a-7120-4b83-b94f-791cf8974ddd](https://github.com/microsoft/azure_arc/assets/12524870/d2c25ad1-57e6-4fac-8cfe-811ebe7b309c)

**Reply by dkirby-ms** [reply]
We recently moved the CDN and thus the images are not yet cached at the various edge locations. This issue should resolve over time.
